app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue

```diff
@@ -174,6 +189,21 @@ export const widgetDefinition = defineWidget(WidgetInput.isAstOrPlaceholder, {
     return Score.Mismatch
   },
 })
+
+/** Custom item added to dropdown. These items can’t be selected, but can be clicked. */
+export interface CustomDropdownItem {
+  /** Displayed label. */
+  label: string
+  /** Action to perform when clicked. */
+  onClick: () => void
+}
+
+export const CustomDropdownItemsKey: unique symbol = Symbol('CustomDropdownItems')
+declare module '@/providers/widgetRegistry' {
+  export interface WidgetInput {
+    [CustomDropdownItemsKey]?: CustomDropdownItem[]
```

Comment:
To avoid doing the same mistake of modifying the input array, consider making this a `ReadonlyArray`. I'm not exactly sure if it will play well with every legitimate usage, but it's worth a try.

Reply:
Seems to be working, done.